sys/net/application_layer/sock_dns/dns.c

```diff
@@ -59,8 +65,10 @@ int sock_dns_query(const char *domain_name, void *addr_out, int family)
         res = sock_udp_recv(&sock_dns, dns_buf, sizeof(dns_buf), 1000000LU, NULL);
         if (res > 0) {
             if (res > (int)DNS_MIN_REPLY_LEN) {
+                uint32_t ttl;
                 if ((res = dns_msg_parse_reply(dns_buf, res, family,
-                                               addr_out)) > 0) {
+                                               addr_out, &ttl)) > 0) {
+                    sock_dns_cache_add(domain_name, addr_out, res, ttl);
```

Comment:
I guess calling `sock_dns_cache_add` within `dns_msg_parse_reply` wouldn't make sense, right?

Reply:
I don't see how - `dns_msg_parse_reply` does exactly what it says - parse the reply.
What happens with the reply is then the job of the calling function.